BLSQ/iaso
# TRIA-117 Allow superuser to export mobile app setup for a user

This PR adds a feature that allows a superuser to download a .zip file for a given user and project.
This zip file contains all the files needed to do the complete setup of the mobile app. It contains the user's profile, an access token and .json files for all resources such as the org unit pyramid and the entities, based on the enabled feature flags.

## Self proofreading checklist

- ~My migrations file are included~

## Doc
- Tell us where the doc can be found (docs folder, wiki, in the code...)

## Changes

- Added an action (icon button) on the users list if the authorized user is a super user
- Generating the export will:
  1. Launch a Task
  2. Poll the task for it's status and progress
  3. When the task has completed successfully, request a signed url (S3) for the generated file
  4. Display the download button
- The task will call all API endpoints and write the responses to files in a temporary directory. When done it adds all files to an archive and uploads to S3.

## How to test

1. Set `DNS_DOMAIN` Env variable to your local machine. Note: I had to use `ngrok` to get it to work because it didn't find my localhost from inside the docker container. Using `ngrok` was the faster solution.
2. Launch the tasks_worker:
  ```
  docker-compose run --rm iaso bash
  python manage.py tasks_worker
  ```
3. Make sure you're a super user and use the dashboard to generate an export

## Print screen / video



https://github.com/BLSQ/iaso/assets/2719739/4774f879-a986-4714-9eaf-584e173e3ec3
